tests/draft4/keyword-independence.json

```diff
@@ -0,0 +1,4397 @@
+                    "x" : 1
+                },
+                "description" : "object, minProperties valid",
+                "valid" : true
+            }
+        ]
+    },
+    {
+        "description" : "contains + patternProperties",
+        "schema" : {
+            "contains" : true,
+            "patternProperties" : {
+                "no" : {
+                    "not" : {}
+                },
+                "yes" : {}
+            }
+        },
+        "tests" : [
+            {
+                "data" : [],
+                "description" : "array, contains invalid",
+                "valid" : false
+            },
+            {
+                "data" : [
+                    1
+                ],
+                "description" : "array, contains valid",
+                "valid" : true
+            },
+            {
+                "data" : {
+                    "no" : 1
+                },
+                "description" : "object, patternProperties invalid",
+                "valid" : false
+            },
+            {
+                "data" : {
+                    "yes" : 1
+                },
+                "description" : "object, patternProperties valid",
+                "valid" : true
+            }
+        ]
+    },
+    {
+        "description" : "contains + properties",
+        "schema" : {
+            "contains" : true,
+            "properties" : {
+                "x" : {
+                    "not" : {}
+                }
+            }
+        },
+        "tests" : [
+            {
+                "data" : [],
+                "description" : "array, contains invalid",
+                "valid" : false
+            },
+            {
+                "data" : [
+                    1
+                ],
+                "description" : "array, contains valid",
+                "valid" : true
+            },
+            {
+                "data" : {
+                    "x" : 1
+                },
+                "description" : "object, properties invalid",
+                "valid" : false
+            },
+            {
+                "data" : {},
+                "description" : "object, properties valid",
+                "valid" : true
+            }
+        ]
+    },
+    {
+        "description" : "contains + propertyNames",
+        "schema" : {
+            "contains" : true,
+            "propertyNames" : {
+                "not" : {}
+            }
+        },
+        "tests" : [
+            {
+                "data" : [],
+                "description" : "array, contains invalid",
+                "valid" : false
+            },
+            {
+                "data" : [
+                    1
+                ],
+                "description" : "array, contains valid",
+                "valid" : true
+            },
+            {
+                "data" : {
+                    "x" : 1
+                },
+                "description" : "object, propertyNames invalid",
```

Comment:
I believe draft4 doesn't have propertyNames.

Reply:
agreed, it's listed as added from draft4->draft6 - https://json-schema.org/draft-06/json-schema-release-notes.html